test/contract_tests.js

```diff
@@ -0,0 +1,498 @@
+        it("[BUG] Revert, If the reportRoot argument is not the merkel root format", async function () {
+            await expect(Notary.connect(Triager).submit(Bytes32_String)); 
+            await expect(Notary.connect(Triager).submit(Bytes32_Zero)).to.be.reverted;
+        })
+
+        it("Revert, If the same report root submitted more than once",async function(){
+            await expect(Notary.connect(Triager).submit(reportRoot))
+            await expect(Notary.connect(Triager).submit(reportRoot)).revertedWith("Bug Report Notary: Report already submitted"); // since report already exists in `reports` mapping.
+        })
+    });
+
+    describe("===> Attest()", function () {
+        let reportRoot,
+            key,
+            commitment;
+
+        beforeEach(async () => {
+            await Notary.connect(Triager).submit(makeSubmitArgs(report1));
+            [reportRoot, key, commitment] = makeAttestArgs(report1, Triager.address);
+        });
+
+        it("Attest the report", async function () {
+            let resp = await Notary.connect(Triager).attest(reportRoot, key, commitment);
+            let block = await ethers.provider.getBlock(resp.blockNumber);
+            expect(resp).to.emit(Notary, "ReportAttestation").withArgs(Triager.address,reportRoot,key,block.timestamp);
+        })
+
+        it("Revert, if the caller is not an `OPERATOR`", async function () {
+            await expect(Notary.connect(Reporter1).attest(reportRoot, key, commitment))
+                .to.be.reverted;
+        })
+
+        it("Revert, if attest more than once on the same report",async function(){
+            await expect(Notary.connect(Triager).attest(reportRoot, key, commitment))
+            await expect(Notary.connect(Triager).attest(reportRoot, key, commitment)).to.be.reverted;
+        })
+
+        it("Revert, if attest `commitment` is empty",async function(){
+            await expect(Notary.connect(Triager).attest(reportRoot, key, Bytes32_Zero)).revertedWith("Bug Report Notary: Invalid commitment");
+        })
+
+        it("Revert, if attest on non-exisiting report",async function(){
+            [reportRoot, key, commitment] = makeAttestArgs(report2,Reporter2.address);
+            await expect(Notary.connect(Triager).attest(reportRoot, key, commitment)).revertedWith("Bug Report Notary: Report not yet submitted");
+        })
+
+    })
+
+    describe("===> getBalance()", function () {
+        let reportRoot,
+            Balance;
+
+        beforeEach(async () => {
+            reportRoot = makeSubmitArgs(report1);
+            await Notary.connect(Triager).submit(reportRoot);
+        });
+
+        it("Should return the amount of tokens deposited to a given report",async function(){
+            Balance = await Escrow.connect(Triager).getBalance(reportRoot, NativeAsset)
+            await expect(ethers.utils.formatEther(Balance)).to.equal('0.0') // since we didn't paid the report, the initial balance of reporter address is 0
```

Comment:
You can use `.to.be.equal` or `to.be.closeTo` on the bignum instead of the string. Not super important, though